New Panoramax Features

mattiapezzotti opened the pull request:
New Year Slider to easily filter images
New Shortcut to hide active streetlevel layers
Many bug fixes and removal of some outdated and useless code

mattiapezzotti committed:
let's start from here

mattiapezzotti committed:
year slider 2.0

mattiapezzotti committed:
year slider

mattiapezzotti committed:
fixed year slider

mattiapezzotti committed:
bug fixes

mattiapezzotti committed:
streetlevel toggle shortcut

mattiapezzotti committed:
added test

mattiapezzotti committed:
merged with develop origin

mattiapezzotti committed:
fixed year slider with shortcut

mattiapezzotti committed:
added slider for every provider, now shortcut greys out checkboxes

mattiapezzotti committed:
merged with origin

mattiapezzotti committed:
npm fixes

mattiapezzotti committed:
hfov, pitch and direction is now held between sequences and images as asked in #10392

mattiapezzotti committed:
fix for #10361 (only panoramax)

mattiapezzotti committed:
lint fixes

tyrasd committed:
restore accidental changes from development branch (unrelated to panoramax)

mattiapezzotti committed:
reverted changes to touch.js

mattiapezzotti committed:
now hiding layers show what was enabled

mattiapezzotti committed:
added export function to match others

mattiapezzotti committed:
fixed a bug which caused the year slider to not function when selection all images, implemented a way to keep track of photo types after refresh and updated the plane_photo.js class to simplify the zoom events and fixed a bug where zooming could bring you out of bounds

mattiapezzotti committed:
added tests

mattiapezzotti committed:
added jsdoc

mattiapezzotti committed:
fixed a bug that prevented mouseup event from firing when using the pannellum frame

Pujakumari1202 commented:
Hi there,

I am Puja Kumari currently planning to contribute to open source through GSOC and I found the OpenStreetMap community particularly inspiring because of its mission to create and provide free, accessible geographic data for the world.

It would be really great if the community members would help me to know regarding how should I get started.

Thanks and Regards

tyrasd committed:
add ticks for existing photos on slider, misc refactorings

tyrasd committed:
Merge branch 'develop' into panoramax_dev

tyrasd committed:
update tests

tyrasd committed:
fix rounding errors when calculating date from slider value

tyrasd committed:
rudimentary support for toDate in date slider

only when iD is started with a "to" date in the hash parameter: show a second slider to visualize and set the "to" date.

also: flip orientation on right-to-left locales

tyrasd committed:
add to changelog

tyrasd committed:
lint

tyrasd committed:
fix date range ticks for kartaview service

tyrasd committed:
Merge branch 'develop' into panoramax_develop

tyrasd committed:
tweak keyboard shortcut description

tyrasd committed:
add to changelog
